### PR TITLE
Improve Rpi wireless config, jsc#SLE-10603

### PR DESCRIPTION
**Pull request by gmoro**
Improve Rpi wireless config, jsc#SLE-10603

Implements ability to repeat configuration in case of failure

### CODE REVIEW

File: files/usr/lib/jeos-firstboot [hunk 1]
```diff
@@ -122,8 +122,11 @@ d(){
 	retval=
 	while true
 	do
+        # disable exit on error for capturing dialog return codes
+	        set +e
```
Comment (Vogtinator) [inline files/usr/lib/jeos-firstboot]:
```
retval=0
dialog --backtitle "$PRETTY_NAME" --output-fd 3 "$@" 3>"${dialog_out}" || retval=$?
```

File: files/usr/share/jeos-firstboot/raspberrywifi [hunk 2]
```diff
@@ -122,23 +120,40 @@ WIRELESS_ESSID='$wlan_network'
...
+			echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
+		fi
 
-	run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+		run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+		d --infobox $"Connecting to wireless network ..." 3 38 || true
+
+		run ifdown $wlan_device &>/dev/null || true
+		# Disabling exit on error to capture exit code
+		set +e
```
Comment (Vogtinator) [inline files/usr/share/jeos-firstboot/raspberrywifi]: Bash 101:

```
if ifup $wlan_device &>/dev/null; then
...
fi
```

File: files/usr/share/jeos-firstboot/raspberrywifi [hunk 3]
```diff
@@ -122,23 +120,40 @@ WIRELESS_ESSID='$wlan_network'
...
+		ifup $wlan_device &>/dev/null
+		ret_val=$?
+		set -e
+		if [ $ret_val -ne 0 ]; then
+			if dialog --yesno $"Connection failed, do you wish to retry?" 0 0; then
```
Comment (Vogtinator) [inline files/usr/share/jeos-firstboot/raspberrywifi]: Is it possible to get an error message from wicked? This error message is pretty useless

File: files/usr/share/jeos-firstboot/raspberrywifi [hunk 4]
```diff
@@ -122,23 +120,40 @@ WIRELESS_ESSID='$wlan_network'
...
+				continue
+			else
+				# Otherwise, return 0 to continue main execution
+				return 0
```
Comment (Vogtinator) [inline files/usr/share/jeos-firstboot/raspberrywifi]: Please use `break` and `return 0` consistently

File: files/usr/share/jeos-firstboot/raspberrywifi [hunk 5]
```diff
@@ -122,23 +120,40 @@ WIRELESS_ESSID='$wlan_network'
...
+			fi
+		else
+			break
+		fi
+	done
 }
 
 # This is called after the configuration steps finished successfully
 raspberrywifi_post()
```
Comment (Vogtinator) [inline files/usr/share/jeos-firstboot/raspberrywifi]: Just remove this?

File: files/usr/lib/jeos-firstboot (hunk 1 above)
Comment (gmoro) [reply files/usr/lib/jeos-firstboot]: Stylistic choice right?

File: files/usr/share/jeos-firstboot/raspberrywifi (hunk 2 above)
Comment (gmoro) [reply files/usr/share/jeos-firstboot/raspberrywifi]: again, it's broke down the way it is on purpose, to make it clearer,  pure stylistic choice.

File: files/usr/share/jeos-firstboot/raspberrywifi (hunk 3 above)
Comment (gmoro) [reply files/usr/share/jeos-firstboot/raspberrywifi]: Sadly no, wicked does not have more then a setup-in-progress message, and wpa_supplicant uses a D-bus api to give status, will be more complicated, but I will check on how to get proper errors later.

File: files/usr/share/jeos-firstboot/raspberrywifi (hunk 4 above)
Comment (gmoro) [reply files/usr/share/jeos-firstboot/raspberrywifi]: I do not understand what is inconsistent here, can you please clarify?

File: files/usr/share/jeos-firstboot/raspberrywifi (hunk 5 above)
Comment (gmoro) [reply files/usr/share/jeos-firstboot/raspberrywifi]: I think is best to leave it so this code can serve as a model for any other "module" for jeos-firstboot, I can actually document the mechanism as well, put some comments maybe.

File: files/usr/share/jeos-firstboot/raspberrywifi (hunk 2 above)
Comment (Vogtinator) [reply files/usr/share/jeos-firstboot/raspberrywifi]: No, this is simply bad code.

File: files/usr/lib/jeos-firstboot (hunk 1 above)
Comment (Vogtinator) [reply files/usr/lib/jeos-firstboot]: Could be argued, but changing the shell mode should be avoided.

File: files/usr/share/jeos-firstboot/raspberrywifi (hunk 4 above)
Comment (Vogtinator) [reply files/usr/share/jeos-firstboot/raspberrywifi]: You use `break` below to achieve the same.

File: files/usr/share/jeos-firstboot/raspberrywifi (hunk 5 above)
Comment (Vogtinator) [reply files/usr/share/jeos-firstboot/raspberrywifi]: That should be documented on https://github.com/openSUSE/jeos-firstboot/wiki/Support-for-extensions-using-custom-scripts

**Pull request merged**